pyspecProcScripts/integrate_limits.py

```diff
@@ -1,43 +1,132 @@
+    cut off from the maximum signal intensity.
 
-    Parameters
-    ==========
-    axis: str
-        name of the dimension along which you want to determine the integration
-        limits
+    axisname: str
+        apply convolution along `axisname`
+
+    fwhm: int
+        width of the matched filter is `fwhm` - this is calculated more precisely
+        within the program
+
+    Gaussian: boolean
+        set `Gaussian` to True in order to calculate the matched-filter via
+        Gaussian convolution of the frequency domain signal (i.e., apply a matched
+        Gaussian filter to peak in frequency domain).
+        If `Gaussian` is set to True, set `Lorentzian` to False.
+
+    Lorentzian: boolean
+        set `Lorentzian` to True in order to calculate the matched-filter via
+        Lorentzian convolution of the frequency domain signal (i.e., apply a
+        matched Lorentzian filter to peak in frequency domain).
+        If `Lorentzian` is set to True, set `Gaussian` to False.
+
+
+    Lorentzian_to_Gaussian: boolean
+        set `Lorentzian_to_Gaussian` to True in order to apply this
+        transformation to your data, in order to determine integral limits.
+        Must set `Lorentzian` to True and `Gaussian` to False.
+        
+    fl: None or figlist_var()
+        to show diagnostic plots, set `fl` to the figure list; set `fl` to None in
+        order not to see any diagnostic plots
 
-    Returns
-    =======
-    retval: ndarray
-        A len 2 ndarray tuple with the start and stop indeces for the integration
-        bounds.
     """
-    signal_sign = s.C.sum(axis).run(np.real).run(np.sign)
-    temp = abs(s).real * signal_sign
-
-    # pulled from apodization code
-    sigma = nddata(np.linspace(1e-5,1e3,1000),'sigma').set_units('sigma','s')
-    s_avg = s.C.mean_all_but('t2')
-    gaussians = np.exp(-s_avg.C.fromaxis('t2')**2/2/sigma**2)
-    signal_E = (abs(s_avg * gaussians)**2).sum('t2')
-    signal_E /= signal_E.data.max()
-    filter_width = abs(signal_E-1/sqrt(2)).argmin('sigma').item()
+    temp = s.C.mean_all_but(axis)
     if fl is not None:
+        fl.next('integration diagnostic')
         fl.push_marker()
-        fl.next('signal Energy')
+        fl.plot(abs(temp.C.ft('t2'))/abs(temp.C.ft('t2')).max(), alpha=0.6, label='before convolve')
+    Gaussian_Conv = Gaussian
+    Lorentzian_Conv = Lorentzian
+    Lorentz_to_Gauss = Lorentzian_to_Gaussian
+    #{{{ make sure to calculate Lorentzian filter for L-to-G transform
+    if Lorentz_to_Gauss:
+        print("Must set Lorentzian convolve to True for Lorentzian to Gaussian transformation... doing so now")
+        Lorentzian_Conv = True
+        Gaussian_Conv = False
+    #}}}
+    # Calculating matched filter
+    sigma = nddata(np.linspace(1e-10,1e-1,1000),'sigma').set_units('sigma','s')
+    if Gaussian_Conv:
+        gaussians = np.exp(-temp.C.fromaxis(axis)**2/2/sigma**2)
+        signal_E = (abs(temp * gaussians)**2).sum(axis)
```

Comment:
this and line 76 can be condensed if you just call "gaussians" and "lorentzians" the same thing (defined differently based on the filter that you're using)

Reply:
I have done this too